From 9fba8655633c058a46910f5afc9fe39ff552e056 Mon Sep 17 00:00:00 2001
From: Quoc Khanh Le <QuocKhanh.Le@arm.com>
Date: Wed, 19 Apr 2023 14:50:49 +0100
Subject: [PATCH 12/13] Disable SMMU stage2 and enable stage1

Use the intergrated MMU600 for the dpu, This avoids issue with the import of buffers at dpu for HW rendering

Signed-off-by: Quoc Khanh Le <QuocKhanh.Le@arm.com>
---
 fdts/tc.dts | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/fdts/tc.dts b/fdts/tc.dts
index 9cf1411c9..23dc333a3 100644
--- a/fdts/tc.dts
+++ b/fdts/tc.dts
@@ -5,7 +5,8 @@
  */
 
 /dts-v1/;
-
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/interrupt-controller/irq.h>
 / {
 	compatible = "arm,tc";
 	interrupt-parent = <&gic>;
@@ -213,10 +214,6 @@
 			linux,cma-default;
 		};
 
-		optee@0xf8e00000 {
-			compatible = "restricted-dma-pool";
-			reg = <0x00000000 0xf8e00000 0 0x00200000>;
-		};
 
 		fwu_mm@0xfca00000 {
 			reg = <0x00000000 0xfca00000 0 0x00400000>;
@@ -480,7 +477,6 @@
 		interrupt-names = "JOB", "MMU", "GPU";
 		clocks = <&gpu_clk>, <&gpu_core_clk>;
 		clock-names = "clk_mali", "shadercores";
-		iommus = <&smmu_700 0x200>;
 		operating-points = <
 			/* KHz uV */
 			50000 820000
@@ -498,13 +494,17 @@
 		ts = <20000 2000 (-20) 2>;
 		thermal-zone = "";
 	};
-
-	smmu_700: smmu_700@3f000000 {
-		#iommu-cells = <1>;
+        
+	mmu_600_tcu: mmu_600_tcu@2ce00000 {
 		compatible = "arm,smmu-v3";
-		reg = <0x0 0x3f000000 0x0 0x5000000>;
-		dma-coherent;
-	};
+		reg = <0 0x2ce00000 0 0x100000>;
+		interrupts = <GIC_SPI 75 IRQ_TYPE_EDGE_RISING>,
+			<GIC_SPI 76 IRQ_TYPE_EDGE_RISING>,
+			<GIC_SPI 77 IRQ_TYPE_EDGE_RISING>;
+                interrupt-names = "eventq", "cmdq-sync", "gerror";
+                //dma-coherent;
+                #iommu-cells = <1>;
+        };
 
 	dp0: display@2cc00000 {
 		#address-cells = <1>;
@@ -512,10 +512,10 @@
 		compatible = "arm,mali-d71";
 		reg = <0 0x2cc00000 0 0x20000>;
 		interrupts = <0 69 4>;
+		iommus = <&mmu_600_tcu 0>;
 		interrupt-names = "DPU";
 		clocks = <&scmi_clk 0>;
 		clock-names = "aclk";
-		iommus = <&smmu_700 0x100>;
 		pl0: pipeline@0 {
 			reg = <0>;
 			clocks = <&scmi_clk 1>;
-- 
2.34.1

